test/junit-formatter.bats

```diff
@@ -103,8 +102,7 @@ TESTSUITES_REGEX="<testsuites time=\"$FLOAT_REGEX\">"
 }
 
 @test "junit formatter as report formatter creates report.xml" {
-  make_bats_test_suite_tmpdir
-  cd "$BATS_TEST_SUITE_TMPDIR" # don't litter sources with output files
+  cd "$BATS_TEST_TMPDIR" # don't litter sources with output files
```

Comment:
I know it is not really part of this PR but could you also take care of the `tab in filename` test? It litters source with the transformed file and should be trivially fixable with the new tmpdirs.

Reply:
I don't get what you would like to see to be done.
If it is not related directly to this PR, why not address this in another one ?
Always better to split PR for different purpose

Reply:
See my new comment. I think it is very fitting for this PR.